spec/support/mail_matcher.rb

```diff
@@ -13,17 +15,40 @@
   end
 end
 
-RSpec::Matchers.define(:include_in_html_body) do |expected|
-  # ignore \r and \n in the comparison
-  mail_body_stripped = ->(mail) { mail.html_part.body.decoded.gsub(/[\r\n]/, "") }
-  expected_stripped = expected.gsub(/[\r\n]/, "")
+RSpec::Matchers.define(:include_in_html_body) do |expected_text|
+  @body_doc = nil
 
   match do |mail|
-    mail_body_stripped.call(mail).include?(expected_stripped)
+    return false unless mail&.html_part&.body
+
+    # Parse the document once and store the Nokogiri object.
+    # The .to_s is kept for clarity and safety, ensuring a string is passed to Nokogiri.
+    @body_doc = Nokogiri::HTML(mail.html_part.body.to_s)
+    body_text = @body_doc.text
+
+    # Normalize whitespace in both strings before comparison.
+    normalized_body = body_text.squish
+    normalized_expected = expected_text.squish
+
+    normalized_body.include?(normalized_expected)
+  end
+
+  failure_message do
+    if @body_doc
+      "Expected the HTML body's text to include:\n  " \
+        "\"#{expected_text.squish}\"\n\n" \
+        "But the parsed text was:\n  " \
+        "\"#{@body_doc.text.squish}\""
+    else
+      "Expected email to have an HTML part, but it was nil."
+    end
   end
 
-  failure_message do |mail|
-    "Expected that the HTML body would include:\n#{expected_stripped}\n" \
-      + "But got:\n#{mail_body_stripped.call(mail)}"
+  failure_message_when_negated do
+    parsed_text = @body_doc ? @body_doc.text.squish : ""
```

Comment:
When there is no `@body_doc`, the `parsed_text` is equal to `""`. In this case, the error message reads weird since it states _"But it was found in the parsed text"_, which is an empty string. Maybe catch this case by its own sentence beforehand.